packages/ui/src/components/OverlayTrigger.tsx

```diff
@@ -0,0 +1,161 @@
+     */
+    props?: Omit<PopoverProps, 'children'>;
+    /**
+     * The content to display in the popover body.
+     */
+    body: React.ReactNode;
+    /**
+     * Optional header content for the popover.
+     */
+    header?: React.ReactNode;
+  };
+  tooltip?: {
+    /**
+     * Additional props to pass to the Tooltip component. `id` is required for accessibility.
+     */
+    props: Omit<TooltipProps, 'children' | 'id'> & { id: string };
+    /**
+     * The content to display in the tooltip body.
+     */
+    body: React.ReactNode;
+  };
+  /**
+   * Whether to trap focus inside the overlay when it's shown.
+   * If true, focus will be trapped and moved to the first focusable element.
+   * @default true
+   */
+  trapFocus?: boolean;
+  /**
+   * Whether to return focus to the trigger element when the overlay is hidden.
+   * @default true
+   */
+  returnFocus?: boolean;
+}
+
+/**
+ * A wrapper around react-bootstrap's OverlayTrigger that adds accessibility features:
+ * - Automatic focus trapping when the overlay is shown
+ * - Auto-focus on the first focusable element in the overlay
+ * - Returns focus to the trigger element when the overlay is hidden
+ * - Automatically constructs a Popover with proper ref management
+ *
+ * This component provides a simpler API than react-bootstrap's OverlayTrigger by
+ * handling the Popover construction and ref management internally.
+ *
+ * @example
+ * ```tsx
+ * <OverlayTrigger
+ *   tooltip={{
+ *     body: 'Tooltip content',
+ *     props: { id: 'tooltip-id' },
+ *   }}
+ *   placement="right"
+ * >
+ *   <button>Hover me</button>
+ * </OverlayTrigger>
+ * ```
+ *
+ * @example
+ * ```tsx
+ * <OverlayTrigger
+ *   popover={{
+ *     header: 'Popover title',
+ *     body: 'Popover content',
+ *   }}
+ *   placement="right"
+ * >
+ *   <button>Click me</button>
+ * </OverlayTrigger>
+ * ```
+ */
+export function OverlayTrigger({
+  children,
+  popover,
+  tooltip,
+  trapFocus: shouldTrapFocus = true,
+  returnFocus = true,
+  onEntered,
+  onExit,
+  ...props
+}: OverlayTriggerProps) {
+  const overlayBodyRef = useRef<HTMLDivElement>(null);
+  const focusTrapRef = useRef<FocusTrap | null>(null);
+  const triggerElementRef = useRef<HTMLElement | null>(null);
+
+  const handleEntered = (node: HTMLElement, isAppearing: boolean) => {
+    // Store the currently focused element (the trigger) before we move focus
+    if (returnFocus && document.activeElement instanceof HTMLElement) {
+      triggerElementRef.current = document.activeElement;
+    }
+
+    if (shouldTrapFocus && overlayBodyRef.current) {
+      // Trap focus inside the overlay body
+      focusTrapRef.current = trapFocus(overlayBodyRef.current);
+
+      // Move focus to the first focusable element
+      focusFirstFocusableChild(overlayBodyRef.current);
+    }
+
+    // Call the original onEntered callback if provided
+    onEntered?.(node, isAppearing);
+  };
+
+  const handleExit = (node: HTMLElement) => {
```

Comment:
This won't run if the component unmounts while it's open, if I understand correctly? Do we need to deactivate the focus trap in a `useEffect` cleanup to catch this case?

Reply:
If it unmounts, the useEffect cleanup wouldn't run, the component isn't mounted

Reply:
Done in [c9d2a9f](https://github.com/PrairieLearn/PrairieLearn/pull/13404/commits/c9d2a9fab604af40af0ed0b9ed3cd2290a717e0c).